### DISCUSSION TITLE
GFF or BEDfile for Roslin Cg genome RepeatMasker output

### DISCUSSION THREAD

**Discussion by yaaminiv** [RobertsLab/resources#1202] [Q&A] [answered]
## How do I convert RepeatMasker output to be something useable with `bedtools`?

I downloaded the RepeatMasker output from NCBI for the Roslin *C. gigas* assembly from [this link](https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/902/806/645/GCF_902806645.1_cgigas_uk_roslin_v1/GCF_902806645.1_cgigas_uk_roslin_v1_rm.out.gz). When I looked at the file, there are a handful of columns that I likely don't need for `bedtools intersect`. 

![Screen Shot 2021-05-05 at 4 05 46 PM](https://user-images.githubusercontent.com/22335838/117220283-c1fcec80-adbb-11eb-9e7b-cc1cfc8f7096.png)

I tried using the existing RepeatMasker output with `intersectBed` just to see if it would even accept the file as is. It did not:

![Screen Shot 2021-05-05 at 4 12 16 PM](https://user-images.githubusercontent.com/22335838/117220732-aa723380-adbc-11eb-99cc-82e42d1fb7c4.png)

I wanted to extract the chr/start/end columns for a BEDfile, but I'm unable to do so! 

![Screen Shot 2021-05-05 at 4 07 38 PM](https://user-images.githubusercontent.com/22335838/117220410-05575b00-adbc-11eb-8ded-b81ce44cd521.png)

In the past @kubu4 has provided GFF RepeatMasker output for *C. virginica*

<img width="1157" alt="Screen Shot 2021-05-05 at 3 58 43 PM" src="https://user-images.githubusercontent.com/22335838/117220843-e7d6c100-adbc-11eb-9987-7c0d632dc00b.png">

**Comment by kubu4** [comment, marked as answer]
Convert to BED file:

`tail -n +4 GCF_902806645.1_cgigas_uk_roslin_v1_rm.out | awk 'BEGIN{OFS= "\t"} {print $5, $6, $7}' > GCF_902806645.1_cgigas_uk_roslin_v1_rm.bed`

The `tail -n +4` skips the first four lines.

EDITED: Hit `Answer` before I was finished.

**Reply by kubu4** [reply]
And, I just happened to find this on the RepeatMasker webpage (was only there to get the URL for the software, but happened to notice this):

>In addition, we have included a useful python tool (RM2Bed.py) developed by David Ray's lab for manipulating/filtering RM annotation files ( *.out ) and saving output to the BED file format. See the RepeatMasker page for installation details.

That's for versions 4.1.0 and up. We have v4.1.0 installed on Mox if you want to give it a shot.

**Comment by yaaminiv** [comment]
> tail -n +4 GCF_902806645.1_cgigas_uk_roslin_v1_rm.out | awk 'BEGIN{OFS= "\t"} {print $5, $6, $7}' > GCF_902806645.1_cgigas_uk_roslin_v1_rm.bed

That worked. Thanks!

http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_rm.te.bed
